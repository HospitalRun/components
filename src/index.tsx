import { library } from '@fortawesome/fontawesome-svg-core'
import { fas } from '@fortawesome/free-solid-svg-icons'

import '../scss/main.scss'

library.add(fas)
export * from './components/Toaster'
export * from './components/Spinner'
export * from './components/Graph'
export * from './components/Button'
export * from './components/Dropdown'
export * from './components/Icon'
export * from './components/Image'
export * from './components/Badge'
export * from './components/Panel'
export * from './components/Pill'
export * from './components/Checkbox'
export * from './components/Navbar'
export * from './components/TextField'
export * from './components/Switch'
export * from './components/Radio'
export * from './components/Alert'
export * from './components/Modal'
export * from './components/TextInput'
export * from './components/Select'
export * from './components/RichText'
export * from './components/List'
export * from './components/DateTimePicker'
export * from './components/VideoPlayer'
export * from './components/Layout'
export * from './components/Label'
export * from './components/Breadcrumb'
export * from './components/Tab'
export * from './components/Typeahead'
export * from './components/Calendar'
export * from './components/Table'
