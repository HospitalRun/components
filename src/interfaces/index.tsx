export type ColorVariant =
  | 'primary'
  | 'secondary'
  | 'success'
  | 'danger'
  | 'warning'
  | 'dark'
  | 'light'
  | 'info'

export type ButtonVariant =
  | 'light'
  | 'dark'
  | 'primary'
  | 'secondary'
  | 'success'
  | 'danger'
  | 'warning'
  | 'info'
  | 'link'
  | 'outline-primary'
  | 'outline-secondary'
  | 'outline-success'
  | 'outline-danger'
  | 'outline-warning'
  | 'outline-info'
  | 'outline-dark'
  | 'outline-light'
