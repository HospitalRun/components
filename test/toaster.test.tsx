import * as React from 'react'
import { mount } from 'enzyme'
import { Toaster, Toast } from '../src'

jest.useFakeTimers()

describe('Toaster', () => {
  it('Toast notification renders itself without crashing', () => {
    const toasterWrapper = mount(<Toaster autoClose={3000} hideProgressBar draggable />)
    const toastifyWrapper = toasterWrapper.find('.Toastify')
    expect(toastifyWrapper).toBeTruthy()
  })

  it('correctly renders a info notification', () => {
    const App = () => (
      <div>
        <Toaster />
      </div>
    )

    const wrapper = mount(<App />)
    Toast('success', 'message')
    Toast('info', 'message')
    Toast('warning', 'message')
    Toast('error', 'message')
    Toast('default', 'message') // tests the default case
    jest.runAllTimers()
    expect(wrapper).toMatchSnapshot()
  })
})
