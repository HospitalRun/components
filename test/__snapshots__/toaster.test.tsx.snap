// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toaster correctly renders a info notification 1`] = `ReactWrapper {}`;
